__tests__/app.test.js

```diff
@@ -105,19 +103,55 @@ describe('GET /api/articles', () => {
     })
     test('Returns 404 not found if no topics are found with that id', () => {
             return request(app)
-            .get('/api/article/222')
+            .get('/api/articles/222')
             .expect(404)
             .then((response) => {
                 expect(response.body.msg).toBe('Not Found');
             })
         })
         test('Returns 400 bad request if a non integer is given as article_id', () => {
             return request(app)
-            .get('/api/article/bad')
+            .get('/api/articles/bad')
             .expect(400)
             .then((response) => {
                 expect(response.body.msg).toBe('Bad Request');
             })
         })
     })
 
+    describe('GET /api/articles/:article_id/comments', () => {
+        test('Returns an array of comment objects for the given article_id, containing comment_id, votes, created_at, author (username from users table, body', () => {
+            return request(app)
+            .get('/api/articles/3/comments')
+            .expect(200)
+            .then((response) => {
+                expect(response.body.comments.length).toBe(2);
+                response.body.comments.forEach(comment => {
+                    expect(comment).toMatchObject({
+                        comment_id: expect.any(Number),
+                        votes: expect.any(Number),
+                        created_at: expect.any(String),
+                        author: expect.any(String),
+                        body: expect.any(String),
+                    })
+                })
+            })
+        })
+        test('Returns 404 not found when given an article id that does not exist', () => {
+            return request(app)
+            .get('/api/articles/222/comments')
+            .expect(404)
+            .then((response) => {
+                expect(response.body.msg).toBe('Not Found');
+            })
+        })
+        test('Returns 204 no content when selecting an article with no comments', ()=> {
+            return request(app)
+            .get('/api/articles/2/comments')
+            .expect(204)
+            .then((response) => {
+                expect(response.body).toEqual({});
+            })
+        })
+    })
+
```

Comment:
Also need to test for an invalid article id (400)

Reply:
consider sending a 200 instead of 204 for empty comments as it can cause problems client side